ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -33,41 +32,43 @@
 import java.util.UUID;
 import java.util.zip.GZIPOutputStream;
 
+import static com.microsoft.azure.kusto.ingest.Validation.validateIsNotNull;
+
 class IngestClientImpl implements IngestClient {
 
     private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
     private static final int COMPRESSED_FILE_MULTIPLIER = 11;
     private final ResourceManager resourceManager;
-    private AzureStorageHelper azureStorageHelper;
+    private AzureStorageClient azureStorageClient;
 
     IngestClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         log.info("Creating a new IngestClient");
         Client client = ClientFactory.createClient(csb);
         this.resourceManager = new ResourceManager(client);
-        this.azureStorageHelper = new AzureStorageHelper();
+        this.azureStorageClient = new AzureStorageClient();
     }
 
     IngestClientImpl(ResourceManager resourceManager) {
         log.info("Creating a new IngestClient");
         this.resourceManager = resourceManager;
-        azureStorageHelper = new AzureStorageHelper();
+        azureStorageClient = new AzureStorageClient();
     }
 
-    IngestClientImpl(ResourceManager resourceManager, AzureStorageHelper azureStorageHelper) {
+    IngestClientImpl(ResourceManager resourceManager, AzureStorageClient azureStorageClient) {
         log.info("Creating a new IngestClient");
         this.resourceManager = resourceManager;
-        this.azureStorageHelper = azureStorageHelper;
+        this.azureStorageClient = azureStorageClient;
     }
 
     @Override
     public IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties)
             throws IngestionClientException, IngestionServiceException {
 
         // Argument validation:
-        if (blobSourceInfo == null || ingestionProperties == null) {
-            throw new IllegalArgumentException("blobSourceInfo or ingestionProperties is null");
-        }
+        validateIsNotNull(blobSourceInfo, "blobSourceInfo is null");
```

Comment:
Can the "is null" string be part of the Ensure.IsNotNull?

Reply:
And this should be all along. The validate should receive object and object name.